src/System.Linq/tests/SelectManyTests.cs

```diff
@@ -452,10 +452,21 @@ public void DisposeAfterEnumeration(int sourceLength, int subLength)
             Assert.All(subState, s => Assert.Equal(subLength + 1, s));
             Assert.All(subCollectionDisposed, t => Assert.True(t));
 
-            // Make sure the iterator's enumerator has been disposed properly.
-            Assert.Equal(0, e.Current); // Default value.
-            Assert.False(e.MoveNext());
-            Assert.Equal(0, e.Current);
+            // .NET Core fixes an oversight where we wouldn't properly dispose
+            // the SelectMany iterator. See https://github.com/dotnet/corefx/pull/13942.
```

Comment:
What was the actual fix, that we just zero-out the value returned from Current?  If that's all the fix was, seems like we should revisit that change to decide whether it's really worth departing from the desktop behavior.  If we do keep the difference, this code could be simplified to:
```C#
int expectedCurrent = PlatformDetection.IsFullFramework ? subLength : 0;
Assert.Equal(expectedCurrent, e.Current);
Assert.False(e.MoveNext());
Assert.Equal(expectedCurrent, e.Current);
```